src/status_im/ui/components/image_with_loader.cljs

```diff
@@ -0,0 +1,32 @@
+(ns status-im.ui.components.image-with-loader
+  (:require [reagent.core :as reagent]
+            [quo.design-system.colors :as colors]
+            [status-im.ui.components.react :as react]
+            [status-im.ui.components.icons.icons :as icons]))
+
+(defn- placeholder [props child]
+  (let [{:keys [style]} props]
+    [react/view {:style (merge style
+                               {:align-items :center
+                                :justify-content :center
+                                :background-color colors/gray-lighter})}
+     child]))
+
+(defn image-with-loader [props]
+  (let [{:keys [source style]} props
+        loaded? (reagent/atom nil)
+        error? (reagent/atom false)]
+    (fn []
+      [:<>
+       (when @error?
+         [placeholder {:style style}
+          [icons/icon :main-icons/cancel]])
+       (when-not (or @loaded? @error?)
+         [placeholder {:style style}
+          [react/activity-indicator {:animating true}]])
+       (when (and (or (nil? @loaded?) @loaded?)
```

Comment:
Hi @msuess, nice implementation 👍
1. `(or (nil? @loaded?) @loaded?)` Isn't this will be always true?
2. Why we are using `nil`, why not default to `false`?

Reply:
Good catch, that doesn't make any sense haha. I swear it did at some point 😅 . Gonna remove this and default to `false`.

Reply:
fixed!